### PR TITLE
Add GCP support

### PR DESCRIPTION
**Pull request by ghost**
The only thing i'm not sure of is, are the changes in bokchoi.py. I had to change

```
def deploy(self, *args, **kwargs):
  print('Deploying: ' + self.config.name)
  return self.backend.deploy(path=self.config.path, *args, **kwargs)
```
to
```
def deploy(self):
  print('Deploying: ' + self.config.name)
  return self.backend.deploy(path=self.config.config_path)
```
because self.config.path does not exist and self.backend.deploy does not require args* and kwargs* as params.

### CODE REVIEW

File: bokchoi/gcp/gcp.py [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            'disk_space': gcp.get('DiskSpaceGb', 50)
+        }
+
+    def list_instances(self):
+        """List names of all existing instances"""
+        instances = []
+        result = self.compute.instances().list(
+            project=self.gcp.get('project'),
+            zone=self.gcp.get('zone')).execute()
+        [instances.append(x['name']) for x in result['items']]
```
Comment (TimNooren) [inline bokchoi/gcp/gcp.py]: I think the list comprehension does not add anything here. I would gor for:
`instances = [x['name'] for x in result['items']]`
and remove `instances = []` above it.

File: bokchoi/gcp/gcp.py [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+        :return: Defined Compute Engine configuration
+        """
+        image_response = self.compute.images().getFromFamily(
+            project='debian-cloud', family='debian-8').execute()
+
+        machine_type = "zones/{}/machineTypes/{}".format(
+            self.gcp.get('zone'), self.gcp.get('instance_type'))
+
+        startup_script = open(
+            os.path.join(os.path.dirname(__file__), '../scripts/gcp-startup-script.sh'), 'r').read()
```
Comment (TimNooren) [inline bokchoi/gcp/gcp.py]: Maybe close the file as well:
`with open('...') as _file: \n startup_script = _file.read()`

File: bokchoi/gcp/gcp.py [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+    def wait_for_operation(self, operation):
+        """Method which polls the status of the operations and returns when the
+        operation is completed.
+        :arg operation: a gcp api operation
+        """
+        if operation is None:
+            return
+
+        print('Waiting for operation to finish...')
+        while True:
```
Comment (TimNooren) [inline bokchoi/gcp/gcp.py:202]: There is a retry function in bokchoi.utils that might be useful

File: bokchoi/scripts/gcp-startup-script.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+BUCKET_NAME=$(curl http://metadata/computeMetadata/v1/instance/attributes/bucket_name -H "Metadata-Flavor: Google")
+PACKAGE_NAME=$(curl http://metadata/computeMetadata/v1/instance/attributes/package_name -H "Metadata-Flavor: Google")
+ENTRYPOINT=$(curl http://metadata/computeMetadata/v1/instance/attributes/entry_point -H "Metadata-Flavor: Google")
+
+gsutil cp gs://${BUCKET_NAME}/${PACKAGE_NAME} .
+unzip ${PACKAGE_NAME}
+
+pip3.4 install -r requirements.txt
+python3.4 ${ENTRYPOINT}
```
Comment (TimNooren) [inline bokchoi/scripts/gcp-startup-script.sh]: Does the instance shut down automatically after this? Or do you have to kill it yourself?

File: bokchoi/gcp/gcp.py (hunk 1 above)
Comment (ghost) [reply bokchoi/gcp/gcp.py]: good call fixing it!

File: bokchoi/gcp/gcp.py (hunk 3 above)
Comment (ghost) [reply bokchoi/gcp/gcp.py:202]: all GCP operations are async and this function blocks the current process until the operation has been succeeded (e.g. make the create instance operation blocking). I wouldn't say that it's the same thing as retrying an operation due to Errors (as described in the utils.retry function).

File: bokchoi/gcp/gcp.py (hunk 2 above)
Comment (ghost) [reply bokchoi/gcp/gcp.py]: added a close()

File: bokchoi/scripts/gcp-startup-script.sh (hunk 4 above)
Comment (ghost) [reply bokchoi/scripts/gcp-startup-script.sh]: it does now :)

**Pull request merged**